src/main/java/dk/kb/image/IIPFacade.java

```diff
@@ -110,58 +112,78 @@ public javax.ws.rs.core.StreamingOutput getIIPImage(
 
         // TODO: Use the UriTemplate system like IIIFFacade
         // http://example.com/iipsrv.fcgi?FIF=/mymount/85/c1/85c1df89-bffe-48e0-8813-111f6f0fba50.jp2&CVT=jpeg
-        UriBuilder builder = UriBuilder.
-                fromUri(ServiceConfig.getConfig().getString(KEY_IIP_SERVER)).
-                queryParam("FIF", FIF); // Mandatory
-        ProxyHelper.addIfPresent(builder, "WID", WID);
-        ProxyHelper.addIfPresent(builder, "HEI", HEI);
-        ProxyHelper.addIfPresent(builder, "RGN", RGN);
-        ProxyHelper.addIfPresent(builder, "QLT", QLT);
-        ProxyHelper.addIfPresent(builder, "CNT", CNT);
+        try {
+            URIBuilder builder = new URIBuilder(ServiceConfig.getConfig().getString(KEY_IIP_SERVER))
```

Comment:
As discussed, I'll switch to `UriTemplate` for `IIPFacade` too. As part of that the `getServer` will be used by both IIIF & IIP.

Reply:
This was quite a large rewrite, primarily because there we no unit tests for the generated URIs.

Unit tests for `IIIFFacade` & `IIPFacade` has now been introduced. The actual proxying of requests has been disabled with [Mockito](https://site.mockito.org/) and the URIs are collected for verification. During this work a few old errors in the argument validation code was caught.